### PR TITLE
Issue 20 - Add validation in url field

### LINKED ISSUE #20: Validate url field
We need to add a validator to the ``url`` field to avoid issues like the one described on #11.

See http://pastebin.com/nLpQZVW3 for more information.

### PR DESCRIPTION
**Pull request by rodfersou**
closes #20

### CODE REVIEW

**Comment by rodfersou** [comment]
I didn't add the translations because it will create problems when merge #14

File: CHANGES.rst [hunk 1]
```diff
@@ -12,6 +12,9 @@ There's a frood who really knows where his towel is.
 - Drop support for Plone 4.1 and remove dependency on unittest2.
   [hvelarde]
 
+- Add validation in URL field (closes `#20`_).
```
Comment (hvelarde) [inline CHANGES.rst]: changelog entries always on top of the others.

File: CHANGES.rst [hunk 2]
```diff
@@ -78,3 +81,4 @@ There's a frood who really knows where his towel is.
 - Initial release.
 
 .. _`#3`: https://github.com/collective/collective.cover/issues/3
+.. _`#20`: https://github.com/collective/collective.cover/issues/20
```
Comment (hvelarde) [inline CHANGES.rst]: sorry, I made the first mistake here: it should be ``https://github.com/simplesconsultoria/sc.embedder/issues/20``

File: src/sc/embedder/content/embedder.py [hunk 3]
```diff
@@ -216,19 +217,27 @@ def get_fallback(self, url):
...
-            consumer = component.getUtility(IConsumer)
-            json_data = consumer.get_data(url, maxwidth=None, maxheight=None,
-                                          format='json')
+        if url == '':
+            return
+
+        v = validation.validatorFor('isURL')
+        if v(url) != 1:
+            plone_utils = api.portal.get_tool(name='plone_utils')
+            plone_utils.addPortalMessage(_(u'URL invalid'), 'info')
```
Comment (hvelarde) [inline src/sc/embedder/content/embedder.py]: you can use just ``api.portal.show_message()``

see: http://docs.plone.org/external/plone.api/docs/portal.html#portal-show-message-example

File: src/sc/embedder/content/embedder.py [hunk 4]
```diff
@@ -216,19 +217,27 @@ def get_fallback(self, url):
 
     def load_oembed(self, action):
         url = self.widgets['url'].value
-        if url != '':
-            consumer = component.getUtility(IConsumer)
-            json_data = consumer.get_data(url, maxwidth=None, maxheight=None,
-                                          format='json')
+        if url == '':
```
Comment (hvelarde) [inline src/sc/embedder/content/embedder.py]: you can just skip it, the validator will take care of this case.

**Comment by hvelarde** [comment]
we need a test for the validation to avoid test coverage decrease.

**Pull request merged**